java/Mekhmet-Efe_Adzhar/src/com/zoolatech/lecture2/tasks/_2/Number2.java

```diff
@@ -0,0 +1,229 @@
+
+                    }
+                    else  {
+                        robotGame.turnRight();
+                        System.out.println("Turned right");
+                    }
+                    break;
+                case 3:
+                    robotGame.getCoordinates();
+                    break;
+                case 4:
+                    robotGame.getDirection();
+                    break;
+                case 5:
+                    System.out.println("Shutting down");
+                    exit = true;
+                    break;
+                default:
+                    break;
+            }
+        }
+    }
+}
+
+enum Direction {
+    SOUTH,
+    NORTH,
+    WEST,
+    EAST
+}
+
+class RobotGame {
+
+    private int w;
+    private int h;
+    private int x;
+    private int y;
+    private int[] coordinates = {x, y};
+    int[][] roomSize;
+
+    private Direction direction = Direction.SOUTH;
+
+    public void setW(int w) {
+        this.w = w;
+    }
+
+    public void setH(int h) {
+        this.h = h;
+    }
+
+    public void getRoomSize() {
+        roomSize = new int[h][w];
+        for (int[] row : roomSize) {
+            for (int i : row) {
+                System.out.print(i);
+            }
+            System.out.println();
+        }
+    }
+
+
+    public void setDirection(Direction direction) {
+        this.direction = direction;
+    }
+
+    public void getCoordinates() {
+        System.out.println(Arrays.toString(coordinates));
+    }
+
+    public void getDirection() {
+        System.out.println(direction);
+    }
+
+    public void setCoordinates(int x, int y) {
+        this.x = x;
+        this.y = y;
+        this.coordinates = new int[]{x, y};
+    }
+
+    public void moveForward() {
+        if (direction == Direction.SOUTH) {
+            if (y < h) {
+                y += 1;
+                setCoordinates(x, y);
+            }
+            if (y == h) {
+                System.out.println("WARNING. OUT OF RANGE. TURN LEFT");
+            }
+        }
+        if (direction == Direction.NORTH) {
+            if (y > 0) {
+                y -= 1;
+                setCoordinates(x, y);
+            } else if (y <= 0) {
+                System.out.println("WARNING. OUT OF RANGE");
+            }
+        }
+        if (direction == Direction.WEST) {
+            if (x > 0) {
+                x -= 1;
+                setCoordinates(x, y);
+            }
+            if (x <= 0) {
+                System.out.println("WARNING. OUT OF RANGE");
+            }
+        }
+        if (direction == Direction.EAST) {
+            if (x < w) {
+                x += 1;
+                setCoordinates(x, y);
+            }
+            if (x >= w) {
+                System.out.println("WARNING. OUT OF RANGE");
+            }
+        }
+    }
+
+    public void turnLeft() {
+        if (direction == Direction.SOUTH) {
+            direction = Direction.EAST;
+            setDirection(direction);
```

Comment:
setDirection does the same as direction = Direction.EAST; (and other directions)

We don't need to perform this operation twice